import/popups/WebrtcPermissionDialog.qml

```diff
@@ -0,0 +1,159 @@
+                source: "image://theme/icon-m-browser-camera"
+            }
+            Icon {
+                visible: "microphone" in devices
+                source: "image://theme/icon-m-browser-microphone"
+            }
+        }
+
+        Label {
+            id: title
+            wrapMode: Text.WordWrap
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+            horizontalAlignment: Text.AlignHCenter
+            font.family: Theme.fontFamilyHeading
+            font.pixelSize: Theme.fontSizeHuge
+            color: Theme.highlightColor
+        }
+
+        Label {
+            id: label
+            wrapMode: Text.WordWrap
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+            horizontalAlignment: Text.AlignHCenter
+            font.pixelSize: Theme.fontSizeSmall
+            color: Theme.highlightColor
+        }
+
+        ComboBox {
+            visible: "camera" in devices
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+
+            //% "Camera"
+            label: qsTrId("sailfish_components_webview_popups-la-webrtc_camera_title")
+
+            menu: ContextMenu {
+                Repeater {
+                    model: devices["camera"]
+                    delegate: MenuItem {
+                        text: translate(model.modelData)
+                    }
+                }
+            }
+            onCurrentIndexChanged: choices["camera"] = currentIndex
+        }
+
+        ComboBox {
+            visible: "microphone" in devices
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+
+            //% "Microphone"
+            label: qsTrId("sailfish_components_webview_popups-la-webrtc_microphone_title")
+
+            menu: ContextMenu {
+                Repeater {
+                    model: devices["microphone"]
+                    delegate: MenuItem {
+                        text: translate(model.modelData)
+                    }
+                }
+            }
+            onCurrentIndexChanged: choices["microphone"] = currentIndex
+        }
+
+        TextSwitch {
+            id: remember
+
+            //: Remember decision for this site for later use
+            //% "Remember for this site"
+            text: qsTrId("sailfish_components_webview_popups-remember_for_site")
+        }
+    }
+
+    Component.onCompleted: {
+        if ("microphone" in devices && "camera" in devices) {
+            choices = {"microphone": 0, "camera": 0}
+            //% "Camera and microphone"
+            title.text = qsTrId("sailfish_components_webview_popups-la-webrtc_camera_microphone_title")
+            //: %1 is the site that wants to use camera and microphone
+            //% "Allow %1 to use your camera and microphone?"
+            label.text = qsTrId("sailfish_components_webview_popups-la-camera_microphone_request")
+                                .arg(host)
```

Comment:
This will mislead about permanent permissions.  I'm thinking about `host: this.uri.host ? this.uri.host : this.uri.filePath` on the embedlite side. The request should be rejected if embedlite fails to provide something readable here, what do you think?

Reply:
What is the effect of making a permanent exception for an empty host in practice? It would allow for more access than just the single file I think, or am I wrong about that? Essentially, I agree the wording needs to reflect exactly the permissions being granted.

I also agree, if there's something odd about the request then better to just reject it. However, granting access for use by a locally stored file sounds like something we should probably support.

Reply:
The permanent permission is stored with an uri, no matter, file or host in the web. So if you give the permission to `webrtc.html`, it is only valid for this particular file. However, (I'm thinking loud) what if the file has changed since the permission was granted? Perhaps we should not give permanent permissions to files unless they come from a secure context.